File: cp_rust/src/bin/abc402-b-restaurant-queue.rs
// Created by Ayush Biswas at 2026/07/08 13:05
// https://atcoder.jp/contests/adt_easy_20250903_1/tasks/abc402_b
use cp_lib::*;

// @code begin
use cpio::*;
use std::collections::VecDeque;

sol! {
    fn solution(
        n: usize,
        queries: [[usize]; n]
    ) -> Lines<usize> {
        let mut queue = VecDeque::new();
        let mut res = vec![];
        for q in queries {
            if q[0] == 1 {
                queue.push_back(q[1]);
            } else if q[0] == 2 {
                res.push(queue.pop_front().unwrap());
            }
        }
        res.into()
    }
}
// @code end

File: cp_nim/src/bin/abc402-b-restaurant-queue.nim
# Created by Ayush Biswas at 2026/07/19 08:29
# https://atcoder.jp/contests/adt_easy_20250903_1/tasks/abc402_b
import ../lib/cpio

# @code begin

import deques

var r = newReader()

let
  n = r[int]
  queries = r[seq[seq[int]], n]

var queue = initDeque[int]()
for q in queries:
  if q[0] == 1:
    queue.addLast q[1]
  else:
    echo queue.popFirst()

# @code end
